pkg/grpc/dapr/dapr_api.go

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2021 Layotto Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dapr
+
+import (
+	"context"
+	"errors"
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/components-contrib/pubsub"
+	"github.com/dapr/components-contrib/state"
+	jsoniter "github.com/json-iterator/go"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/anypb"
+	"mosn.io/layotto/components/configstores"
+	"mosn.io/layotto/components/file"
+	"mosn.io/layotto/components/hello"
+	"mosn.io/layotto/components/lock"
+	runtime_common "mosn.io/layotto/components/pkg/common"
+	"mosn.io/layotto/components/rpc"
+	mosninvoker "mosn.io/layotto/components/rpc/invoker/mosn"
+	"mosn.io/layotto/components/sequencer"
+	grpc_api "mosn.io/layotto/pkg/grpc"
+	dapr_common_v1pb "mosn.io/layotto/pkg/grpc/dapr/proto/common/v1"
+	"mosn.io/layotto/pkg/grpc/dapr/proto/runtime/v1"
+	dapr_v1pb "mosn.io/layotto/pkg/grpc/dapr/proto/runtime/v1"
+	"mosn.io/layotto/pkg/messages"
+	mgrpc "mosn.io/mosn/pkg/filter/network/grpc"
+	"mosn.io/pkg/log"
+	"strings"
+)
+
+type DaprGrpcAPI interface {
+	dapr_v1pb.DaprServer
+	grpc_api.GrpcAPI
+}
+
+type daprGrpcAPI struct {
+	appId                    string
+	hellos                   map[string]hello.HelloService
+	configStores             map[string]configstores.Store
+	rpcs                     map[string]rpc.Invoker
+	pubSubs                  map[string]pubsub.PubSub
+	stateStores              map[string]state.Store
+	transactionalStateStores map[string]state.TransactionalStore
+	fileOps                  map[string]file.File
+	lockStores               map[string]lock.LockStore
+	sequencers               map[string]sequencer.Store
+	sendToOutputBindingFn    func(name string, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error)
+	// app callback
+	AppCallbackConn *grpc.ClientConn
+	// json
+	json jsoniter.API
+}
+
+func (d *daprGrpcAPI) Init(conn *grpc.ClientConn) error {
+	// 1. set connection
+	d.AppCallbackConn = conn
+	return d.startSubscribing()
+}
+
+func (d *daprGrpcAPI) startSubscribing() error {
+	// TODO
```

Comment:
It's better to record what to do here, in case no one would know what to do here in the future